apps/04TowersOfHanoi.js

```diff
@@ -10,36 +10,71 @@ var stacks = {
     c: []
 };
 
+var moveCount = 0;
+
 function printStacks() {
     console.log("a: " + stacks.a);
     console.log("b: " + stacks.b);
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function restartGame() {
+    moveCount = 0;
+    var stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+    };
+    return stacks;
+}
 
+function movePiece(startStack, endStack) {
+    if (isLegal(startStack,endStack)) {
```

Comment:
The movePiece function should only move the piece. Logic to determine whether or not the piece should be moved or not should be done by whatever function is calling the movePiece function. Functions should be kept as modular and ''one-purpose" as possible.